crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -212,26 +231,19 @@ where
             }
         }
 
-        self.update(self.algorithm_updater.algorithm()).await;
-        Ok(())
-    }
-
-    async fn apply_da_block_costs_to_gas_algorithm(
-        &mut self,
-        da_block_costs: DaBlockCosts,
-    ) -> anyhow::Result<()> {
-        self.handle_da_block_costs(da_block_costs).await?;
-        self.update(self.algorithm_updater.algorithm()).await;
         Ok(())
     }
 }
 
 #[async_trait]
-impl<L2, Metadata, DA> RunnableTask for GasPriceServiceV1<L2, Metadata, DA>
+impl<L2, DA, StorageTxProvider> RunnableTask
+    for GasPriceServiceV1<L2, DA, StorageTxProvider>
 where
     L2: L2BlockSource,
-    Metadata: MetadataStorage,
     DA: DaBlockCostsSource,
+    StorageTxProvider: TransactionableStorage + 'static,
```

Comment:
Probably we don't need `'static`

Reply:
Nice catch. There might be other places too since I kinda spread the love around in an earlier iteration.